app/Http/Controllers/Api/EventController.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Rogue\Http\Controllers\Api;
+
+use Rogue\Models\Event;
+use Illuminate\Http\Request;
+use Rogue\Http\transformers\EventTransformer;
+
+class EventController extends ApiController
+{
+    /**
+     * Create a controller instance.
+     */
+    public function __construct()
+    {
+        $this->transformer = new EventTransformer();
+    }
+
+    /**
+     * Returns events.
+     * GET /events
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $query = $this->newQuery(Event::class);
+        $filters = $request->query('filter');
+
+        if ($filters['signup_id']) {
+            $events = $query->where('eventable_id', $filters['signup_id'])->orderBy('created_at', 'desc')->get();
+
+            $signupEvents = [];
+
+            foreach ($events as $event) {
+                if ($event->eventable_type === "Rogue\Models\Signup") {
+                    array_push($signupEvents, $event);
+                }
+            }
+
+            return $this->collection($signupEvents);
```

Comment:
We probably still want to paginate this, since otherwise a signup with a lot of events could break the query (even if we decide not to expose a pagination interface in the admin interface for now).

Reply:
@DFurnes how do we grab all of the results if the pagination is on?

Reply:
We don't want to! (Or if we do, we'd want to add pagination on the front-end).

Reply:
ooohhhh I see! So is this bad then too? https://github.com/DoSomething/rogue/blob/master/app/Http/Controllers/Api/ActivityController.php#L43 (we wanted to show all user activity instead of just the first 20 that were returned - we probably want to paginate this on the front end?

Reply:
Yeah, we always want to make sure we have a reasonable upper-bound on response size. Otherwise edge-cases with particularly "busy" posts (or a malicious user who just requested `/v2/activity?limit=all` 😱) could hammer our server.

Reply:
And for a MVP, I don't think it's unreasonable to say "here are the latest 20" history events, or something of that sort. We can always add pagination later if people request it.

Reply:
Makes total sense - thanks for explaining! Made a card for it here: https://www.pivotaltracker.com/n/projects/2019429/stories/151566633 :)

Reply:
Totally my fault for suggesting this “all” option. @dfunes thanks!

Reply:
Not a big deal at all! 😄

Reply:
I'm glad I learned this @sbsmith86 so it was a good mistake!

Reply:
@dfurnes updated!